elifetools/tests/test_parse_jats.py

```diff
@@ -1849,6 +1849,145 @@ def test_references_json(self, filename):
         ]
         ),
 
+         # example of data citation with a pub-id pub-id-type="archive", parse it as an accession number, based on 00666 kitchen sink example
+        ('''<root xmlns:xlink="http://www.w3.org/1999/xlink"><article><back><ref-list>    <ref id="bib34">
+        <element-citation publication-type="data">
+            <person-group person-group-type="author">
+                <name>
+                    <surname>Radoshevich</surname>
+                    <given-names>L</given-names>
+                </name>
+                <name>
+                    <surname>Impens</surname>
+                    <given-names>F</given-names>
+                </name>
+                <name>
+                    <surname>Ribet</surname>
+                    <given-names>D</given-names>
+                </name>
+                <name>
+                    <surname>Quereda</surname>
+                    <given-names>JJ</given-names>
+                </name>
+                <name>
+                    <surname>Nam Tham</surname>
+                    <given-names>T</given-names>
+                </name>
+                <name>
+                    <surname>Nahori</surname>
+                    <given-names>MA</given-names>
+                </name>
+                <name>
+                    <surname>Bierne</surname>
+                    <given-names>H</given-names>
+                </name>
+                <name>
+                    <surname>Dussurget</surname>
+                    <given-names>O</given-names>
+                </name>
+                <name>
+                    <surname>Pizarro-Cerdá</surname>
+                    <given-names>J</given-names>
+                </name>
+                <name>
+                    <surname>Knobeloch</surname>
+                    <given-names>KP</given-names>
+                </name>
+                <name>
+                    <surname>Cossart</surname>
+                    <given-names>P</given-names>
+                </name>
+            </person-group>
+            <year iso-8601-date="2015">2015a</year>
+            <data-title>ISG15 counteracts <italic>Listeria monocytogenes</italic>
+                infection</data-title>
+            <source>ProteomeXchange</source>
+            <pub-id pub-id-type="archive"
+                xlink:href="http://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD001805"
+                >PXD001805</pub-id>
+        </element-citation>
+    </ref></ref-list></back></article></root>''',
+        [
+            OrderedDict([
```

Comment:
I know that this is the test pattern followed in this project, but these data parameters/fixtures make the tests harder to follow and would benefit from being broken out into separate fixtures, especially as its really only xml and json being passed in.

For the xml they could just be stored in files and parsed, and to keep order when parsing a json file (that seems to be required here) just use `json.load(json_file, object_pairs_hook=OrderedDict)` to keep the order and return an `OrderedDict` instead of a `dict`.

It probably would require a separate PR but thought I would mention it.

Reply:
Yes, thanks for your patience. Changing the tests is in my todo list, and I will probably start on it this month. I don't want it to hold up the bug fix here.